pkg/syndesis/action/checkupdates.go

```diff
@@ -0,0 +1,49 @@
+package action
+
+import (
+	"github.com/operator-framework/operator-sdk/pkg/sdk"
+	"github.com/sirupsen/logrus"
+	"github.com/syndesisio/syndesis-operator/pkg/apis/syndesis/v1alpha1"
+	"github.com/syndesisio/syndesis-operator/pkg/syndesis/version"
+)
+
+// Checks if the syndesis installation should be upgraded and move to the "Upgrading" status.
+type CheckUpdates struct {
+	operatorVersion	string
+}
+
+func (a *CheckUpdates) CanExecute(syndesis *v1alpha1.Syndesis) bool {
+	return syndesisInstallationStatusIs(syndesis, v1alpha1.SyndesisInstallationStatusInstalled)
+}
+
+func (a *CheckUpdates) Execute(syndesis *v1alpha1.Syndesis) error {
+
+	if a.operatorVersion == "" {
+		operatorVersion, err := version.GetSyndesisVersionFromOperatorTemplate()
+		if err != nil {
+			return err
+		}
+		a.operatorVersion = operatorVersion
+	}
+
+	namespaceVersion, err := version.GetSyndesisVersionFromNamespace(syndesis.Namespace)
+	if err != nil {
+		return err
+	}
+
+	if namespaceVersion == a.operatorVersion {
+		// Everything fine
+		return nil
+	} else {
+		// Let's start the upgrade process
+		target := syndesis.DeepCopy()
+		target.Status.InstallationStatus = v1alpha1.SyndesisInstallationStatusUpgrading
+		target.Status.Reason = v1alpha1.SyndesisStatusReasonMissing
```

Comment:
Maybe we could add here the original and target version ?

Reply:
Yes, I was confused myself when I recorded the demo :smile: